### PR TITLE
nomad: 0.9.5 -> 0.10.0

### PR DESCRIPTION
**Pull request by endocrimes**
###### Motivation for this change

Updates Nomad to the recently released 0.10, and also adds myself as a maintainer after handling many of the recent Nomad releases.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [na] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @rushmorem @pradeepchhetri

### CODE REVIEW

File: pkgs/applications/networking/cluster/nomad/default.nix [hunk 1]
```diff
@@ -1,8 +1,8 @@
-{ stdenv, buildGoPackage, fetchFromGitHub }:
+{ stdenv, buildGo112Package, fetchFromGitHub }:
 
-buildGoPackage rec {
+buildGo112Package rec {
```
Comment (jonringer) [inline pkgs/applications/networking/cluster/nomad/default.nix]: can you do:
```
    buildGoPackage = buildGo112Package;
```
in top-level/all-packages.nix please :), then revert the change in the package :)

Comment (endocrimes) [reply pkgs/applications/networking/cluster/nomad/default.nix]: As a specific override for this package? Or?

Comment (jonringer) [reply pkgs/applications/networking/cluster/nomad/default.nix]:
```
  nomad = callPackage ../applications/networking/cluster/nomad { 
    buildGoPackage = buildGo112Package;
  };
```

Comment (endocrimes) [reply pkgs/applications/networking/cluster/nomad/default.nix]: ah - thanks!

Comment (endocrimes) [reply pkgs/applications/networking/cluster/nomad/default.nix]: Done!

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
executable seems to work

```
[3 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/71699
1 package were build:
nomad
```
